### DISCUSSION TITLE
ValueError: File format not supported: filepath=/root/.cache/doctr/models/parseq-24cf693e/weights. Keras 3 only supports V3 .keras and .weights.h5 files, or legacy V1/V2 .h5 files.

### DISCUSSION THREAD

**Discussion by abhijeetsourav** [mindee/doctr#1726] [Q&A]
### Bug description

when i try to fine tune a pretrained parseq model on a custom dataset, I got the following error
**How to fine tune a model on a custom dataset?**
I want to generate a .pb file for running the model on edge devices?
So, I tried to save the model using **model.save(<path/to/save>.keras)**......
I don't know how to proceed further,
**please help me to get the model file for running the parseq optimized model on edge devices, like android.....**

### Code snippet to reproduce the bug

```
!python3 references/recognition/train_tensorflow.py parseq --train_path <path/to/train_dataset> --val_path <path/to/eval_dataset> --epochs 1 --pretrained
```

### Error traceback

```
Traceback (most recent call last):
  File "/content/doctr/references/recognition/train_tensorflow.py", line 447, in <module>
    main(args)
  File "/content/doctr/references/recognition/train_tensorflow.py", line 180, in main
    model = recognition.__dict__[args.arch](
  File "/content/doctr/doctr/models/recognition/parseq/tensorflow.py", line 505, in parseq
    return _parseq(
  File "/content/doctr/doctr/models/recognition/parseq/tensorflow.py", line 481, in _parseq
    load_pretrained_params(model, default_cfgs[arch]["url"])
  File "/content/doctr/doctr/models/utils/tensorflow.py", line 73, in load_pretrained_params
    model.load_weights(f"{params_path}{os.sep}{internal_name}")


  File "/usr/local/lib/python3.10/dist-packages/keras/src/utils/traceback_utils.py", line 122, in error_handler
    raise e.with_traceback(filtered_tb) from None
  File "/usr/local/lib/python3.10/dist-packages/keras/src/saving/saving_api.py", line 268, in load_weights
    raise ValueError(
ValueError: File format not supported: filepath=/root/.cache/doctr/models/parseq-24cf693e/weights. Keras 3 only supports V3 `.keras` and `.weights.h5` files, or legacy V1/V2 `.h5` files.
```

### Environment

WARNING: All log messages before absl::InitializeLog() is called are written to STDERR
I0000 00:00:1725892688.189874    9133 cuda_executor.cc:1015] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355
I0000 00:00:1725892688.197677    9133 cuda_executor.cc:1015] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355
I0000 00:00:1725892688.197925    9133 cuda_executor.cc:1015] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355
DocTR version: 0.9.1a0
TensorFlow version: 2.17.0
PyTorch version: 2.4.0+cu121 (torchvision 0.19.0+cu121)
OpenCV version: 4.10.0
OS: Ubuntu 22.04.3 LTS
Python version: 3.10.12
Is CUDA available (TensorFlow): Yes
Is CUDA available (PyTorch): Yes
CUDA runtime version: 12.2.140
GPU models and configuration: GPU 0: Tesla T4
Nvidia driver version: 535.104.05
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.8.9.6
/usr/lib/x86_64-linux-gnu/libcudnn_adv_infer.so.8.9.6
/usr/lib/x86_64-linux-gnu/libcudnn_adv_train.so.8.9.6
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_infer.so.8.9.6
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_train.so.8.9.6
/usr/lib/x86_64-linux-gnu/libcudnn_ops_infer.so.8.9.6
/usr/lib/x86_64-linux-gnu/libcudnn_ops_train.so.8.9.6

### Deep Learning backend

is_tf_available: True
is_torch_available: True

**Comment by felixdittrich92** [comment]
Hi @abhijeetsourav 👋,
Ref.: #1461 
We don't support Keras v3 atm please downgrade to 
https://github.com/mindee/doctr/blob/9045dcfc9c5c837b06fcda8e802f7cf1d95bd18c/pyproject.toml#L57
and Keras v2 :)

**Comment by abhijeetsourav** [comment]
@felixdittrich92 
Thanks for the quick reply...
please tell me how to convert the saved weights to **.pb** file for use in edge devices like android....
more specifically, I want to convert parseq model to run on android device....

**Comment by felixdittrich92** [comment]
You can try:
```
import tf 
from doctr.models import parseq
model = parseq(pretrained=True, exportable=True)
tf.saved_model.save(model, "parseq.pb")
```

Our Focus is more on Onnx here you can check out:
https://github.com/felixdittrich92/OnnxTR

Best,
Felix :)
